### ISSUE TITLE
Does anyone runs the code successfully?

### ISSUE THREAD

**Issue by KeepNoob** [wl-zhao/DiffSwap#29]
I cannot run the successfully. Perhaps it may be caused by package version conflict. When I create the environment, the following errors have been shown 
```sh
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
altair 5.4.1 requires typing-extensions>=4.10.0; python_version < "3.13", but you have typing-extensions 4.5.0 which is incompatible.
```
If I do `pip install altair==5.4.1`, then the other package error happen.
```sh
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tensorflow 2.13.0 requires typing-extensions<4.6.0,>=3.6.6, but you have typing-extensions 4.12.2 which is incompatible.
```
And notice that the `mtcnn` depends on `tensorflow` package.

**Comment by b4402** [comment]
facing the same issue

**Comment by dsj320** [comment]
我运行成功了，这是我的环境包 name: diffswap
channels:
  - conda-forge
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch/
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main/
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free/
  - defaults
dependencies:
  - _libgcc_mutex=0.1=conda_forge
  - _openmp_mutex=4.5=2_gnu
  - aria2=1.37.0=hbc8128a_2
  - asttokens=2.4.1=pyhd8ed1ab_0
  - backcall=0.2.0=pyh9f0ad1d_0
  - bzip2=1.0.8=h4bc722e_7
  - c-ares=1.34.4=hb9d3cd8_0
  - ca-certificates=2025.1.31=hbcca054_0
  - comm=0.2.2=pyhd8ed1ab_0
  - cuda-cudart=12.6.77=h5888daf_0
  - cuda-cudart_linux-64=12.6.77=h3f2d84a_0
  - cuda-nvrtc=12.6.85=hbd13f7d_0
  - cuda-version=12.6=h7480c83_3
  - cudnn=8.9.7.29=h092f7fd_3
  - debugpy=1.6.7=py38h6a678d5_0
  - decorator=5.1.1=pyhd8ed1ab_0
  - dlib=19.24.6=cuda120py38hf1b9e5f_0
  - executing=2.0.1=pyhd8ed1ab_0
  - importlib-metadata=7.1.0=pyha770c72_0
  - importlib_metadata=7.1.0=hd8ed1ab_0
  - ipykernel=6.29.3=pyhd33586a_0
  - ipython=8.12.2=pyh41d4057_0
  - jedi=0.19.1=pyhd8ed1ab_0
  - jupyter_client=8.6.1=pyhd8ed1ab_0
  - jupyter_core=5.7.2=py38h578d9bd_0
  - ld_impl_linux-64=2.38=h1181459_1
  - libblas=3.9.0=31_h59b9bed_openblas
  - libcblas=3.9.0=31_he106b2a_openblas
  - libcublas=12.6.4.1=h5888daf_1
  - libcurand=10.3.7.77=hbd13f7d_0
  - libcusolver=11.7.1.2=h5888daf_1
  - libcusparse=12.5.4.2=hbd13f7d_0
  - libffi=3.4.4=h6a678d5_0
  - libgcc=14.2.0=h77fa898_1
  - libgcc-ng=14.2.0=h69a702a_1
  - libgfortran=14.2.0=h69a702a_2
  - libgfortran5=14.2.0=hf1ad2bd_2
  - libgomp=14.2.0=h77fa898_1
  - libiconv=1.17=hd590300_2
  - libjpeg-turbo=3.0.0=hd590300_1
  - liblapack=3.9.0=31_h7ac8fdf_openblas
  - liblzma=5.6.4=hb9d3cd8_0
  - libnsl=2.0.1=hd590300_0
  - libnvjitlink=12.6.85=hbd13f7d_0
  - libopenblas=0.3.29=pthreads_h94d23a6_0
  - libpng=1.6.47=h943b412_0
  - libsodium=1.0.18=h36c2ea0_1
  - libsqlite=3.48.0=hee588c1_1
  - libssh2=1.11.1=hf672d98_0
  - libstdcxx=14.2.0=hc0a3c3a_1
  - libstdcxx-ng=14.2.0=h4852527_1
  - libuuid=2.38.1=h0b41bf4_0
  - libxcrypt=4.4.36=hd590300_1
  - libxml2=2.13.5=h0d44e9d_1
  - libzlib=1.3.1=hb9d3cd8_2
  - matplotlib-inline=0.1.6=pyhd8ed1ab_0
  - nccl=2.26.2.1=ha44e49d_1
  - ncurses=6.4=h6a678d5_0
  - nest-asyncio=1.6.0=pyhd8ed1ab_0
  - openssl=3.5.0=h7b32b05_0
  - packaging=24.0=pyhd8ed1ab_0
  - parso=0.8.3=pyhd8ed1ab_0
  - pexpect=4.9.0=pyhd8ed1ab_0
  - pickleshare=0.7.5=py_1003
  - pip=23.3.1=py38h06a4308_0
  - platformdirs=4.2.0=pyhd8ed1ab_0
  - prompt-toolkit=3.0.42=pyha770c72_0
  - prompt_toolkit=3.0.42=hd8ed1ab_0
  - psutil=5.9.8=py38h01eb140_0
  - ptyprocess=0.7.0=pyhd3deb0d_0
  - pure_eval=0.2.2=pyhd8ed1ab_0
  - pygments=2.17.2=pyhd8ed1ab_0
  - python=3.8.18=hd12c33a_1_cpython
  - python-dateutil=2.9.0=pyhd8ed1ab_0
  - python_abi=3.8=2_cp38
  - pyzmq=25.1.2=py38h6a678d5_0
  - readline=8.2=h5eee18b_0
  - setuptools=68.2.2=py38h06a4308_0
  - six=1.16.0=pyh6c4a22f_0
  - sqlite=3.32.3=hcee41ef_1
  - stack_data=0.6.2=pyhd8ed1ab_0
  - tk=8.6.13=noxft_h4845f30_101
  - tornado=6.4=py38h01eb140_0
  - traitlets=5.14.2=pyhd8ed1ab_0
  - wcwidth=0.2.13=pyhd8ed1ab_0
  - wheel=0.41.2=py38h06a4308_0
  - xz=5.4.6=h5eee18b_0
  - zeromq=4.3.5=h6a678d5_0
  - zipp=3.21.0=pyhd8ed1ab_0
  - zlib=1.3.1=hb9d3cd8_2
  - pip:
      - absl-py==2.2.2
      - aiohappyeyeballs==2.4.4
      - aiohttp==3.10.11
      - aiosignal==1.3.1
      - antlr4-python3-runtime==4.9.3
      - astunparse==1.6.3
      - async-timeout==5.0.1
      - attrs==25.3.0
      - cachetools==5.5.2
      - certifi==2025.1.31
      - charset-normalizer==3.4.1
      - clip==0.2.0
      - contourpy==1.1.1
      - cycler==0.12.1
      - easydict==1.13
      - einops==0.8.1
      - filelock==3.16.1
      - flatbuffers==25.2.10
      - fonttools==4.57.0
      - frozenlist==1.5.0
      - fsspec==2025.3.0
      - future==1.0.0
      - gast==0.4.0
      - google-auth==2.39.0
      - google-auth-oauthlib==1.0.0
      - google-pasta==0.2.0
      - grpcio==1.70.0
      - h5py==3.11.0
      - huggingface-hub==0.30.2
      - idna==3.10
      - imageio==2.35.1
      - importlib-resources==6.4.5
      - imutils==0.5.4
      - jinja2==3.1.6
      - keras==2.13.1
      - kiwisolver==1.4.7
      - kornia==0.7.3
      - kornia-rs==0.1.8
      - lazy-loader==0.4
      - libclang==18.1.1
      - lightning-utilities==0.11.9
      - markdown==3.7
      - markupsafe==2.1.5
      - matplotlib==3.7.5
      - mpmath==1.3.0
      - mtcnn==0.1.1
      - multidict==6.1.0
      - networkx==3.1
      - numpy==1.24.3
      - nvidia-cublas-cu12==12.1.3.1
      - nvidia-cuda-cupti-cu12==12.1.105
      - nvidia-cuda-nvrtc-cu12==12.1.105
      - nvidia-cuda-runtime-cu12==12.1.105
      - nvidia-cudnn-cu12==8.9.2.26
      - nvidia-cufft-cu12==11.0.2.54
      - nvidia-curand-cu12==10.3.2.106
      - nvidia-cusolver-cu12==11.4.5.107
      - nvidia-cusparse-cu12==12.1.0.106
      - nvidia-nccl-cu12==2.19.3
      - nvidia-nvjitlink-cu12==12.8.93
      - nvidia-nvtx-cu12==12.1.105
      - oauthlib==3.2.2
      - omegaconf==2.3.0
      - opencv-python==4.11.0.86
      - opt-einsum==3.4.0
      - pillow==10.4.0
      - propcache==0.2.0
      - protobuf==4.25.6
      - pyasn1==0.6.1
      - pyasn1-modules==0.4.2
      - pydeprecate==0.3.1
      - pyparsing==3.1.4
      - pytorch-lightning==1.4.5
      - pywavelets==1.4.1
      - pyyaml==6.0.2
      - regex==2024.11.6
      - requests==2.32.3
      - requests-oauthlib==2.0.0
      - rsa==4.9.1
      - safetensors==0.5.3
      - scikit-image==0.21.0
      - scipy==1.10.1
      - sympy==1.13.3
      - tensorboard==2.13.0
      - tensorboard-data-server==0.7.2
      - tensorflow==2.13.1
      - tensorflow-estimator==2.13.0
      - tensorflow-io-gcs-filesystem==0.34.0
      - termcolor==2.4.0
      - tifffile==2023.7.10
      - tokenizers==0.20.3
      - torch==2.2.0
      - torchmetrics==0.6.0
      - torchvision==0.17.0
      - tqdm==4.67.1
      - transformers==4.46.3
      - triton==2.2.0
      - typing-extensions==4.13.2
      - urllib3==2.2.3
      - werkzeug==3.0.6
      - wrapt==1.17.2
      - yarl==1.15.2

**Comment by hun9008** [comment]
i success in this condition. (Python 3.8.20)

Package                      Version
---------------------------- -----------
absl-py                      0.15.0
aiohappyeyeballs             2.4.4
aiohttp                      3.10.11
aiosignal                    1.3.1
antlr4-python3-runtime       4.9.3
asttokens                    3.0.0
astunparse                   1.6.3
async-timeout                5.0.1
attrs                        25.3.0
backcall                     0.2.0
cachetools                   5.5.2
certifi                      2025.7.14
charset-normalizer           3.4.2
clang                        5.0
contourpy                    1.1.1
cycler                       0.12.1
decorator                    5.2.1
dlib                         20.0.0
easydict                     1.10
einops                       0.6.1
executing                    2.2.0
filelock                     3.16.1
flatbuffers                  1.12
fonttools                    4.57.0
frozenlist                   1.5.0
fsspec                       2025.3.0
ftfy                         6.2.3
future                       1.0.0
gast                         0.4.0
google-auth                  2.40.3
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
grpcio                       1.70.0
h5py                         3.1.0
hf-xet                       1.1.5
huggingface-hub              0.33.4
idna                         3.10
imageio                      2.35.1
importlib_metadata           8.5.0
importlib_resources          6.4.5
imutils                      0.5.4
ipython                      8.12.3
jedi                         0.19.2
Jinja2                       3.1.6
keras                        2.6.0
Keras-Preprocessing          1.1.2
kiwisolver                   1.4.7
kornia                       0.7.0
lazy_loader                  0.4
libclang                     18.1.1
Markdown                     3.7
MarkupSafe                   2.1.5
matplotlib                   3.7.5
matplotlib-inline            0.1.7
mpmath                       1.3.0
mtcnn                        0.1.0
multidict                    6.1.0
networkx                     3.1
numpy                        1.23.5
nvidia-cublas-cu12           12.1.3.1
nvidia-cuda-cupti-cu12       12.1.105
nvidia-cuda-nvrtc-cu12       12.1.105
nvidia-cuda-runtime-cu12     12.1.105
nvidia-cudnn-cu12            9.1.0.70
nvidia-cufft-cu12            11.0.2.54
nvidia-curand-cu12           10.3.2.106
nvidia-cusolver-cu12         11.4.5.107
nvidia-cusparse-cu12         12.1.0.106
nvidia-nccl-cu12             2.20.5
nvidia-nvjitlink-cu12        12.9.86
nvidia-nvtx-cu12             12.1.105
oauthlib                     3.3.1
omegaconf                    2.3.0
openai-clip                  1.0.1
opencv-python                4.12.0.88
opt-einsum                   3.3.0
packaging                    25.0
parso                        0.8.4
pexpect                      4.9.0
pickleshare                  0.7.5
pillow                       10.4.0
pip                          24.2
prompt_toolkit               3.0.51
propcache                    0.2.0
protobuf                     3.20.3
ptyprocess                   0.7.0
pure_eval                    0.2.3
pyasn1                       0.6.1
pyasn1_modules               0.4.2
pycocoevalcap                1.2
pycocotools                  2.0.7
pyDeprecate                  0.3.1
Pygments                     2.19.2
pyparsing                    3.1.4
python-dateutil              2.9.0.post0
pytorch-lightning            1.4.2
PyWavelets                   1.4.1
PyYAML                       6.0.2
regex                        2024.11.6
requests                     2.32.4
requests-oauthlib            2.0.0
rsa                          4.9.1
safetensors                  0.5.3
scikit-image                 0.21.0
scipy                        1.10.1
setuptools                   75.1.0
six                          1.15.0
stack-data                   0.6.3
sympy                        1.13.3
tensorboard                  2.13.0
tensorboard-data-server      0.7.2
tensorflow                   2.6.0
tensorflow-estimator         2.6.0
tensorflow-io-gcs-filesystem 0.34.0
termcolor                    1.1.0
tifffile                     2023.7.10
tokenizers                   0.20.3
torch                        2.4.1
torchmetrics                 0.6.0
torchvision                  0.19.1
tqdm                         4.67.1
traitlets                    5.14.3
transformers                 4.46.3
triton                       3.0.0
typing_extensions            4.13.2
urllib3                      2.2.3
wcwidth                      0.2.13
Werkzeug                     3.0.6
wheel                        0.44.0
wrapt                        1.12.1
yarl                         1.15.2
zipp                         3.20.2
